packages/common/src/utils/filters.ts

```diff
@@ -425,45 +427,57 @@ export const getFilterRulesByFieldType = (
 export const getDashboardFilterRulesForTile = (
     tileUuid: string,
     rules: DashboardFilterRule[],
-): DashboardFilterRule[] =>
+): DashboardFilterRuleForTile[] =>
     rules
         .filter((rule) => !rule.disabled)
-        .map((filter) => {
+        .reduce<DashboardFilterRuleForTile[]>((acc, filter) => {
             const tileConfig = filter.tileTargets?.[tileUuid];
 
             // If the config is false, we remove this filter
             if (tileConfig === false) {
-                return null;
+                return acc;
             }
 
+            const dashboardFilterRuleForTile = filter;
+
+            // Delete properties that are not needed for filters on tiles
+            if ('tileTargets' in dashboardFilterRuleForTile)
+                delete dashboardFilterRuleForTile.tileTargets;
```

Comment:
should avoid mutation, it could change the filters passed in as arguments, did you mean to copy the filter?

`const dashboardFilterRuleFortile = {...filter}` ?

Reply:
oops! Missed that on a refactor - thank you